inference.py

```diff
@@ -43,112 +45,223 @@
+            ############################
+            # Test Implementation of the NIR
+            ############################
+            # Init NIR   TODO: make a proper way to read the NIR channel
+            #                  and put an option to be able to give the idex of the NIR channel
+            # Extract the NIR channel -> [batch size, H, W] since it's only one channel
+            inputs_NIR = inputs[:, -1, ...]
+            # add a channel to get the good size -> [:, 1, :, :]
+            inputs_NIR.unsqueeze_(1)
+            # take out the NIR channel and take only the RGB for the inputs
+            inputs = inputs[:, :-1, ...]
+            # Suggestion of implementation
+            # inputs_NIR = data['NIR'].to(device)
+            inputs = [inputs, inputs_NIR]
+            # outputs = model(inputs, inputs_NIR)
+            ############################
+            # End of the test implementation module
+            ############################
+        output_lst = []
+        for transformer in transforms:
+            # augment inputs
+            augmented_input = transformer.augment_image(inputs)
+            augmented_output = model(augmented_input)
+            if isinstance(augmented_output, OrderedDict) and 'out' in augmented_output.keys():
+                augmented_output = augmented_output['out']
+            logging.debug(f'Shape of augmented output: {augmented_output.shape}')
+            # reverse augmentation for outputs
+            deaugmented_output = transformer.deaugment_mask(augmented_output)
+            deaugmented_output = F.softmax(deaugmented_output, dim=1).squeeze(dim=0)
+            output_lst.append(deaugmented_output)
+        outputs = torch.stack(output_lst)
+        outputs = torch.mul(outputs, WINDOW_SPLINE_2D)
+        outputs, _ = torch.max(outputs, dim=0)
+        outputs = outputs.permute(1, 2, 0)
+        outputs = outputs.reshape(padded, padded, num_classes).cpu().numpy().astype('float16')
+        outputs = outputs[dist_samples:-dist_samples, dist_samples:-dist_samples, :]
+        fp[row:row + chunk_size, col:col + chunk_size, :] = \
+            fp[row:row + chunk_size, col:col + chunk_size, :] + outputs
+        cnt += 1
+    fp.flush()
+    del fp
+
+    fp = np.memmap(tp_mem, dtype='float16', mode='r', shape=(h_, w_, num_classes))
+    subdiv = 2.0
+    step = int(chunk_size / subdiv)
+    pred_img = np.zeros((h_, w_), dtype=np.uint8)
+    for row in tqdm(range(0, input_image.height, step), position=2, leave=False):
+        for col in tqdm(range(0, input_image.width, step), position=3, leave=False):
+            arr1 = fp[row:row + chunk_size, col:col + chunk_size, :] / (2 ** 2)
+            arr1 = arr1.argmax(axis=-1).astype('uint8')
+            pred_img[row:row + chunk_size, col:col + chunk_size] = arr1
+    pred_img = pred_img[:h, :w]
+    end_seg = time.time() - start_seg
```

Comment:
here again, comments would help follow what's going on. Many lines are similar to those in gen_img_sample(). Could this be refactored?

Reply:
most probably, code improvement is always work in progress